pennylane/compiler/compiler.py

```diff
@@ -66,12 +66,12 @@ def _check_compiler_version(name):
 
     if name == "catalyst":
         installed_catalyst_version = metadata.version("pennylane-catalyst")
+        name = name + installed_catalyst_version
```

Comment:
Doesn't this break future checks in this function?

Reply:
Yes goo catch I am changing that

Reply:
What is the motivation behind the change?

Reply:
I've had a look, we seem to only invoke the check in calls to `qml.qjit`, so any potential overhead due to the removal of the cache is likely irrelevant 👍 

Double checking with @maliasadi that it's ok to remove